tests/test_compare_dictconfig_vs_dict.py

```diff
@@ -0,0 +1,211 @@
+"""
+This file compares dunder methods of DictConfig (e.g. __getitem__ and
+__setitem__) with dunder methods of standard python's dict.
+
+We have separate test classes for the following cases:
+    TestUntypedDictConfig: for DictConfig without a set key_type
+    TestPrimitiveTypeDunderMethods: for DictConfig where key_type is primitive
+    TestEnumTypeDunderMethods: for DictConfig where key_type is Enum
+"""
+from enum import Enum
+from typing import Any, Dict
+
+import pytest
+
+from omegaconf import DictConfig
+from omegaconf.errors import KeyValidationError
+from tests import Enum1
+
+
+@pytest.fixture(
+    params=[
+        "a",
+        "b",
+        1,
+        2,
+        123.45,
+        67.89,
+        True,
+        False,
+        Enum1.FOO,
+        Enum1.BAR,
+        "FOO",
+        "BAR",
+    ]
+)
+def key(request: Any) -> Any:
+    """A key to test indexing into DictConfig."""
+    return request.param
+
+
+@pytest.fixture
+def python_dict(data: Dict[Any, Any]) -> Dict[Any, Any]:
+    """Just a standard python dictionary, to be used in comparison with DictConfig."""
+    return data.copy()
```

Comment:
deepcopy is safer.

Reply:
Made the change in https://github.com/omry/omegaconf/pull/484/commits/79893d4305910a0312c7b3e99846ef4e4c1a7fdf